src/datasets/features/features.py

```diff
@@ -295,7 +295,9 @@ def _cast_to_python_objects(obj: Any, only_1d_for_numpy: bool, optimize_list_cas
         import PIL.Image
 
     if isinstance(obj, np.ndarray):
-        if not only_1d_for_numpy or obj.ndim == 1:
+        if obj.ndim == 0:
+            return obj[()], True
```

Comment:
0-dim arrays are not supported in PyArrow by default, so we need to extract the scalar value first while preserving its type ([SO thread](https://stackoverflow.com/questions/9814226/error-extracting-element-from-an-array-python) that explains how this can be done).